sktm/jenkins.py

```diff
@@ -82,13 +86,54 @@ def __get_build(self, job, buildid):
                 return build
             except Exception as e:
                 logging.warning("catch %s: %s" % (type(e), e))
-                logging.info("now sleep 60s and try again")
-                time.sleep(60)
+                logging.info("now sleep %ds and try again" % interval)
+                time.sleep(interval)
 
         logging.error("fail to get build after retry %d times" %
                       self.retry_cnt)
         raise e
 
+    def __get_job_prop(self, job, attr, interval):
+        """
+        Get property of Jenkins job. Retry Jenkins self.retry_cnt times
+        in case of temporary network failures.
+
+        Args:
+            job:        Jenkins job.
+            attr:       Attribute of job.
+            interval:   Seconds to sleep before retrying.
+
+        Return:
+            job property if succeed, else:
+            (a) directly raise AttributeError if the attr is invalid.
+            OR
+            (b) raise the last exception after retry self.retry_cnt times.
+        """
+        try:
```

Comment:
Oh. I though it would be less code to add the exception catching to the existing code. If we get this exception, we raise it right away in the first iteration. There's no need to have a separate try/except block, only the order of catching the exception is enforced.

Reply:
Agree. Use func = getattr(job, method) instead.